LIPS/lip-3.md

```diff
@@ -0,0 +1,152 @@
+- total amount of signing keys submitted is less than or equal to the new staking limit.
+
+#### `TopUpLegoProgram` script factory contract
+
+`TopUpLegoProgram` contract creates EVM Script to top up the address of the LEGO program. Allows transferring various ERC20 tokens and ETH.
+
+#### `AddRewardProgram` script factory contract
+
+`AddRewardProgram` contract creates EVM Script to add new reward program address to `RewardProgramsRegistry`. 
+`RewardProgramsRegistry` contract is an additional contract storing a whitelist of allowed reward program addresses.
+To successfully complete the EVM script generation, the following requirements must be met:
+- `RewardProgramsRegistry` doesn’t already contain reward program address.
+
+#### `RemoveRewardProgram` script factory contract
+
+`RemoveRewardProgram` contract creates EVM Script to remove an existing reward program address from `RewardProgramsRegistry`. 
+To successfully complete the EVM script generation, the following requirements must be met:
+- `RewardProgramsRegistry` already contains the reward program address.
+
+#### `TopUpRewarProgram` script factory contract
+
+`TopUpRewarProgram` contract creates EVM Script to top up balances of reward programs. Addresses to be topped up are restricted to the list stored in the `RewardProgramsRegistry` contract.
+
+## `EVMScriptExecutor` contract
+
+The EVM scripts generated by the script factory contracts can be executed by a separate `EVMScriptExecutor` contract. It implements only `executeEVMScript()` method and accepts scripts built following the Aragon's EVM script structure. Current implementation uses [CallsScript.sol](https://github.com/aragon/aragonOS/blob/next/contracts/evmscript/executors/CallsScript.sol) contract of the Aragon framework. 
+`EVMScriptExecutor` permissions list is the same as the one of Aragon Voting App. While sharing the EVM script structure, both `EasyTrack` contract and Aragon Voting App are able to execute scripts using `EVMScriptExecutor` contract.
+___
+# Sanity and security
+
+The proposed Easy Track Motions design involves timelock as a native security feature. Any suspicious motion can be rejected or canceled within a relatively long motion duration period. To prevent malicious actions, we propose putting the following additional limitations in place (numeric values fixed based on community feedback from v1):
+
+- It should be impossible to set objections threshold at more than 5% of the total LDO supply. Nevertheless, the default objections threshold should be set a lot lower at 0.5% of the total LDO supply.
+- It should be impossible to set motion duration at less than 48 hours. For any motion, there should be enough time for DAO to submit objections and reject the motion. 
+- It should be impossible to spam motions. The default limit for simultaneously active motions will be set at 12 and it can only be increased by the DAO to up to 24 motions at a time.
```

Comment:
Let's add explicitly that the limit is on all types of motions in total, and not for every time in particular?

Reply:
Good point.